source/simulator/core.cc

```diff
@@ -1320,6 +1381,26 @@ namespace aspect
     return ret;
   }
 
+  /**
+   * Split the set of DoFs (typically locally owned or relevant) in @p whole_set into blocks
+   * given by the @p dofs_per_block structure.
+   */
+  void  split_by_block (std::vector<IndexSet> &partitioned,
```

Comment:
By the way, isn't this function exactly the same as the one by same name that's already in namespace `Utilities`?